### ISSUE TITLE
Book opinion: Crafting Interpreters

### ISSUE THREAD

**Issue by guissalustiano** [grammarware/software-evolution#1]
Hey,
Do you have some opinion about the [Crafting Interpreter from Robert Nystrom](https://craftinginterpreters.com/) book?
I'm looking for a book to help me with the compiler implementation in this course and also give me a full view of the field.
Thanks!

**Comment by grammarware** [comment]
On the first sight, seems like quite a reasonable book. I'll bring it together with my opinion to the next lectorial (this Thursday)

**Comment by guissalustiano** [comment]
I will not be able to attend the lectorial on Thursday, please write here too.

**Comment by grammarware** [comment]
Will do!

**Comment by grammarware** [comment]
I ended up spending half of the lectorial on the book, slides can be found on Canvas: https://canvas.utwente.nl/courses/13975/files/4153169
TL/DR version:
- it's fine, so if it resonates with you better than alternatives, go for it, it is a decent option
- one third of the book is spent on writing an interpreter, the other two thirds on writing a completely different processor of the same language, targeting a virtual machine — start with the first part, and it might be enough for you for this course (or for the rest of your life); you can always go back to the second part for fun
- the book has a bit of a hobby feeling: on one hand it's somewhat more enthusiastic than an average alternative, but on the other hand some definitions might not be completely synced with the state of the art
- there are also occasional sweeping remarks that should be taken with a grain of salt, but that goes for all books to some extent
- since I have the book now, you can borrow it from me until the end of the course if you want a physical copy. I have it with me at the lectorial today and can take it with me the next time, otherwise my office is 3102 in Zilverling, you can try randomly dropping by

**Comment by guissalustiano** [comment]
I'm so sad to have missed the class, but the slides and TLDR were enlightening. 
Thanks so much!
